tasks/cluster_agent.py

```diff
@@ -187,10 +187,18 @@ def hacky_dev_image_build(
     target_image="cluster-agent",
     push=False,
     signed_pull=False,
+    arch=None,
 ):
     os.environ["DELVE"] = "1"
     build(ctx)
 
+    if arch is None:
+        arch = CONTAINER_PLATFORM_MAPPING.get(platform.machine().lower())
+
+    if arch is None:
+        print("Unable to determine architecture to build, please set `arch`", file=sys.stderr)
+        raise Exit(code=1)
```

Comment:
nit - would be more Pythonic but feel free to ignore

```suggestion
    if not arch:
        arch = CONTAINER_PLATFORM_MAPPING.get(platform.machine().lower())

    if not arch:
        print("Unable to determine architecture to build, please set `arch`", file=sys.stderr)
        raise Exit(code=1)
```